xbmc/network/Network.cpp

```diff
@@ -206,39 +258,86 @@ bool CNetwork::IsLocalHost(const std::string& hostname)
 
 CNetworkInterface* CNetwork::GetFirstConnectedInterface()
 {
-   std::vector<CNetworkInterface*>& ifaces = GetInterfaceList();
-   std::vector<CNetworkInterface*>::const_iterator iter = ifaces.begin();
-   while (iter != ifaces.end())
-   {
-      CNetworkInterface* iface = *iter;
-      if (iface && iface->IsConnected())
-         return iface;
-      ++iter;
-   }
+   iter_for_each(GetInterfaceList(), CNetworkInterface, iface)
```

Comment:
You want to utilise c++11 here be doing

for (auto &iface : GetInterfaceList())

Thats how you get rid of the macro you mentioned